### PR TITLE
feat: add in-depth analytics

### PR DESCRIPTION
**Pull request by MaiaS**
## WHAT IT DOES:
_Description of the pull request, what changed_
Adds Analytics to site.
Adds a new hook `useAnalytics` and a new file `analytics/events` which holds constants of analytic event names.

New tracking analytics calls have been added for almost every interaction on the site.

## HOW TO TEST:
_Description of how to test if applicable_

Navigate to the page.
Check the network tab.
Check to see that analytics are being fired.

Screen Shot 2023-02-15 at 6 50 03 PM

Alternatively you could install the chrome extension for easy debugging
https://chrome.google.com/webstore/detail/segment-event-tracker/hbanigoffkilibdakdmmlgefndpjmajl?hl=en

## WHAT'S LEFT
- [ ] - Consent manager if necessary
- [x] - GA hook up in Segment
- [ ] - Add share button analytics once implemented

### CODE REVIEW

File: analytics/events.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const EVENT_NAMES = {
+  INPUT: `Input mode selected`,
+  SUBMIT: "Submitted request",
+  LENGTH: `Length selected`,
+  SHARED: "Share button clicked",
+  NEW_SUMMARY: "New summary button clicked",
+};
```
Comment (pratikmdhr) [inline analytics/events.ts:9]: Do you think we should add an event for the Results page (like RESULT_VIEWED) too? that way we can ensure if the users are reaching the results page after the SUBMIT event and not getting stuck in the loading page

**Review by pratikmdhr — APPROVED** [review]
Hey thanks for setting this up so fast, looking great! Added one comment but still good to merge!

File: analytics/events.ts (hunk 1 above)
Comment (MaiaS) [reply analytics/events.ts:9]: Good idea!

Comment (MaiaS) [reply analytics/events.ts:9]: I added the results, I added better error handling and analytics for the errors as well!

Comment (pratikmdhr) [reply analytics/events.ts:9]: @MaiaS 🔥🔥

**Pull request merged**